backend/src/zac_lite/user_tasks/serializers.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass
+from typing import Optional
+from uuid import UUID
+
+from django.utils.encoding import force_bytes
+from django.utils.http import urlsafe_base64_encode
+from django.utils.translation import gettext_lazy as _
+
+from django_camunda.api import get_task
+from django_camunda.camunda_models import Task
+from rest_framework import serializers
+from rest_framework.request import Request
+
+from .tokens import token_generator
+
+FRONTEND_URL = "/ui/perform-task/{tidb64}/{token}"
+
+
+@dataclass
+class UserTaskLink:
+    request: Request
+    task: Optional[Task] = None
+
+    @property
+    def task_id(self):
+        return self.task.id
+
+    @property
+    def url(self):
+        assert self.task is not None, "Expected task to be validated and resolved"
+        tidb64 = urlsafe_base64_encode(force_bytes(self.task.id))
+        token = token_generator.make_token(self.task)
+        ui_path = FRONTEND_URL.format(tidb64=tidb64, token=token)
+        return self.request.build_absolute_uri(ui_path)
+
+
+class UserLinkSerializer(serializers.Serializer):
+    task_id = serializers.UUIDField(
+        required=True,
+        label=_("Camunda Task ID"),
+        help_text=_(
+            "The ID of the user task in Camunda, which is normally in the UUID 4 format."
```

Comment:
_Required_ and _normally_ and _UUIDField_ kind of set off an alarm bell. Will the abnormal cases also work?

Reply:
So far I haven't seen cases where it isn't a UUID. The API documentation of Camunda says it's a string, we observe that the format UUID applies. With their latest version they also specify an OpenAPI documentation, but you need to extract it from their jars it seems so I haven't been able to verify that they also specify it as UUID. We can easily relax it to just CharField if it needs to be though.